BraveRewardsUI/Tipping/TippingViewController.swift

```diff
@@ -72,9 +155,46 @@ class TippingViewController: UIViewController, UIViewControllerTransitioningDele
   }
   
   @objc private func tappedSendTip() {
-    tippingView.setTippingConfirmationVisible(true, animated: true)
-    DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { [weak self] in
-      self?.dismiss(animated: true)
+    if let selectedIndex = self.tippingView.optionSelectionView.selectedOptionIndex {
+      let amount = self.tippingView.optionSelectionView.options[selectedIndex].value.doubleValue
+      
+      state.ledger.listRecurringTips { [weak self] info in
+        guard let self = self else { return }
+        
+        // If the user has recurring tips, remove the monthly tip
+        if info.firstIndex(where: { $0.id == self.publisherInfo.id }) != nil {
+          self.state.ledger.removeRecurringTip(publisherId: self.publisherInfo.id)
+        }
+        
+        // Add recurring tips if there is none..
+        if self.tippingView.optionSelectionView.isMonthly {
+          self.state.ledger.addRecurringTip(publisherId: self.publisherInfo.id, amount: amount)
+        }
+        
+        self.state.ledger.tipPublisherDirectly(self.publisherInfo, amount: Int32(amount), currency: "BAT")
+        
+        let displayConfirmationView = { (recurringDate: String?) in
+          let provider = " \(self.publisherInfo.provider.isEmpty ? "" : String(format: Strings.OnProviderText, self.publisherInfo.provider))"
+          
+          self.tippingView.setInfo(name: "\(self.publisherInfo.name)\(provider)", tipAmount: amount, recurringDate: recurringDate)
+          self.tippingView.setTippingConfirmationVisible(true, animated: true)
+          
+          DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
+            self.dismiss(animated: true)
+          }
+        }
+        
+        if self.tippingView.optionSelectionView.isMonthly {
+          let date = Date(timeIntervalSince1970: TimeInterval(self.state.ledger.autoContributeProps.reconcileStamp))
+          let dateString = DateFormatter().then {
+            $0.dateFormat = "MM/dd/yyyy"
```

Comment:
Do we have to specify date format here? Would be better to have it locale dependent